### ISSUE TITLE
Claim types configuration can't be edited. AzureCP not working for SharePoint 2019

### ISSUE THREAD

**Issue by edise91** [Yvand/AzureCP#142]
Hi there

I cannot edit the claim types configuration. Anyway the people picker doesn't work. I'm trying to setup a SharePoint-2019 On-Premise Farm with AAD SAML SSO and AzureCP as described: https://docs.microsoft.com/en-us/azure/active-directory/saas-apps/sharepoint-on-premises-tutorial

I tried to reset the configuration, but it didn't help. 
I tried to search in people picker, but i can't find any users.
I tried different claims provided in Azure, but it didn't help.
I tried version 18,19 and newest Beta version, but it dind't help.
The connection to azure seems to be fine.
There are no error outputs.

Any suggestions where the problem could be ?

Kind regards

**Comment by Yvand** [comment]
If you followed the article you should not have to edit the claim types configuration.
To understand what is happening, you can look at the SharePoint logs to view the AzureCP activity - as documented in https://azurecp.yvand.net/docs/help/troubleshooting/

**Comment by edise91** [comment]
I wanted to edit the claim types configuration, because the article didn't work for me. I wanted to edit the claim mapping. 
I performed a troubleshooting:

Test connectivity with Azure AD -> connection successfully
Obtain the access token using PowerShell# --> Received Access Token successfully

When I inspect the SharePoint logs i see followeing entries:
1. Configuration 'AzureCPConfig' was upgraded in configuration database and some settings were updated or reset to their default configuration

2. Claim types configuration list was reset because the identity claim type was either not found or not configured correctly

3. [AzureCP] Unexpected error occurred Microsoft.Graph could not query tenant '****.onMicrosoft.com': Microsoft.Graph.ServiceException: Code: Authorization_RequestDenied  Message: Insufficient privileges to complete the operation.  Inner error:   AdditionalData:   date: 2022-01-24T18:03:59   request-id: 8a625a2d-a6e7-4753-845b-492c3ce8070f   client-request-id: 8a625a2d-a6e7-4753-845b-492c3ce8070f  ClientRequestId: 8a625a2d-a6e7-4753-845b-492c3ce8070f  , Callstack:   
 at Microsoft.Graph.HttpProvider.<SendAsync>d__18.MoveNext()  --- End of stack trace from previous location where exception was thrown ---    
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()    
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    
 at Microsoft.Graph.BaseRequest.<SendRequestAsync>d__38.MoveNext()  --- End of stack trace from previous location where exception was thrown ---    
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()    
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    
 at Microsoft.Graph.BaseRequest.<SendAsync>d__34`1.MoveNext()  --- End of stack trace from previous location where exception was thrown ---    
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()    
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    
 at Microsoft.Graph.GraphServiceUsersCollectionRequest.<GetAsync>d__4.MoveNext()  --- End of stack trace from previous location where exception was thrown ---    
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()    
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    
 at azurecp.AzureCP.<>c__DisplayClass60_0.<<QueryAzureADTenantAsync>b__0>d.MoveNext()  --- End of stack trace from previous location where exception was thrown ---    
 at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()    
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    
 at azurecp.AzureCP.<QueryAzureADTenantAsync>d__60.MoveNext()




I also found following article with the AppPoolAccount but it didn't help. Neither Farm Admin, nor Pool Admin nor domain admin.
